File: app/src/main/java/de/christinecoenen/code/zapp/model/SimpleChannelList.java
package de.christinecoenen.code.zapp.model;


import android.support.annotation.NonNull;

import java.util.Iterator;
import java.util.List;

/**
 * Simple IChannelList wrapper around a List of ChannelModels.
 */
public class SimpleChannelList implements IChannelList {

	private final List<ChannelModel> channels;

	public SimpleChannelList(List<ChannelModel> channels) {
		this.channels = channels;
	}

	@Override
	public ChannelModel get(int index) {
		return channels.get(index);
	}

	@Override
	public ChannelModel get(String id) {
		int index = indexOf(id);
		return index == -1 ? null : get(index);
	}

	@Override
	public int size() {
		return channels.size();
	}

	@Override
	public List<ChannelModel> getList() {
		return channels;
	}

	@Override
	public int indexOf(String channelId) {
		for (int i = 0; i < channels.size(); i++) {
			if (channels.get(i).getId().equals(channelId)) {
				return i;
			}
		}
		return -1;
	}

	@NonNull
	@Override
	public Iterator<ChannelModel> iterator() {
		return channels.iterator();
	}
}

File: app/src/main/java/de/christinecoenen/code/zapp/model/SimpleChannelList.kt
package de.christinecoenen.code.zapp.model

/**
 * Simple IChannelList wrapper around a List of ChannelModels.
 */
class SimpleChannelList(private val channels: List<ChannelModel>) : IChannelList {

	override val list: List<ChannelModel>
		get() = channels

	override fun get(index: Int): ChannelModel {
		return channels[index]
	}

	override fun get(id: String): ChannelModel {
		val index = indexOf(id)
		return this[index]
	}

	override fun size(): Int {
		return channels.size
	}

	override fun indexOf(channelId: String): Int {
		return channels.withIndex().find { indexedValue ->
			indexedValue.value.id == channelId
		}!!.index
	}

	override fun iterator(): Iterator<ChannelModel> {
		return channels.iterator()
	}
}
